File: src/backend/modules/m6_groupware/schemas/notices.py
"""
M6 그룹웨어 — 공지사항 Pydantic 스키마
"""
from typing import Optional
from pydantic import BaseModel


class NoticeCreate(BaseModel):
    title: str
    content: str
    is_pinned: bool = False
    is_important: bool = False


class NoticeUpdate(BaseModel):
    title: Optional[str] = None
    content: Optional[str] = None
    is_pinned: Optional[bool] = None
    is_important: Optional[bool] = None


class NoticeListItem(BaseModel):
    id: str
    title: str
    is_pinned: bool
    is_important: bool
    view_count: int
    author_id: str
    author_name: Optional[str] = None
    created_at: Optional[str] = None


class NoticeResponse(NoticeListItem):
    content: str
    updated_at: Optional[str] = None

File: src/frontend/src/api/groupware/notices.ts
/**
 * M6 그룹웨어 — 공지사항 API 호출 함수
 */
import api from '../client';

export interface NoticeListItem {
  id: string;
  title: string;
  is_pinned: boolean;
  is_important: boolean;
  view_count: number;
  author_id: string;
  author_name?: string;
  created_at?: string;
}

export interface NoticeDetail extends NoticeListItem {
  content: string;
  updated_at?: string;
}

export async function listNotices(params?: { page?: number; size?: number }) {
  const res = await api.get('/groupware/notices', { params });
  return res.data?.data ?? res.data;
}

export async function getNotice(id: string) {
  const res = await api.get(`/groupware/notices/${id}`);
  return res.data?.data ?? res.data;
}

export async function createNotice(data: {
  title: string;
  content: string;
  is_pinned?: boolean;
  is_important?: boolean;
}) {
  const res = await api.post('/groupware/notices', data);
  return res.data?.data ?? res.data;
}

export async function updateNotice(id: string, data: {
  title?: string;
  content?: string;
  is_pinned?: boolean;
  is_important?: boolean;
}) {
  const res = await api.put(`/groupware/notices/${id}`, data);
  return res.data?.data ?? res.data;
}

export async function deleteNotice(id: string) {
  const res = await api.delete(`/groupware/notices/${id}`);
  return res.data?.data ?? res.data;
}
